.travis.yml

```diff
@@ -20,6 +19,27 @@ install:
   - rustup target add riscv32imc-unknown-none-elf
   - rustup component add rustfmt
   - rustup component add clippy
+  - cargo install -f elf2tab --version 0.4.0
+  # Ubuntu doesn't package QEMU for RISC-V, so let's build it
+  - wget https://download.qemu.org/qemu-4.2.0.tar.xz
+  - tar xJf qemu-4.2.0.tar.xz
+  - pushd qemu-4.2.0
+  - ./configure --target-list=riscv32-softmmu
+  - make -j8
+  - popd
+  # Build Tock, it needs to be outside of the libtock-rs source
+  - pushd ../
+  - git clone --depth=50 https://github.com/tock/tock.git
+  - cd tock/boards/hifive1
+  # Use a known working version of Tock
+  - git checkout c94059d3e25dc635e682facff4894ef43b9aca0e
```

Comment:
Maybe we could package the kernel as an artifact inside the tock os build. Then we wouldn't have to rebuild it all the time.

Reply:
That would be cool! I can look into that.

Reply:
I have concerns about the `--depth=50` breaking quickly, though those are only relevant if you don't check in the kernel as a build artifact.

If you do check it in, it's probably worth providing the build script you used so we can reproduce the binary.

Reply:
It looks like it isn't simple to get the build artifacts from Tock, we would have to save it to a S3 instance: https://docs.travis-ci.com/user/uploading-artifacts/

I have removed the `--depth 50` as I agree.

Reply:
We could use the github releases flow to provide artifacts. However, for now (as tock releases are done manually) we can just attach kernel binaries to each release until there is an automated workflow. I think the tests should run ideally with the version provided in the Readme to really test compatibility. Now, it's not possible because as I understand there is no tock release supporting riscv userland apps.

Reply:
I agree it would be best to use a released version of Tock binaries, but currently that isn't an option. After the next release we can change to just downloading the release.

Reply:
:+1: